src/services/authService.ts

```
@@ -0,0 +1,105 @@
+
+const authBase = () => buildApiUrl('zeneyer-auth/v1');
+
+const parseJsonOrThrow = async (res: Response): Promise<unknown> => {
+  const text = await res.text();
+  if (!res.ok) {
+    // Try to parse as JSON for a structured error message, fall back to HTTP status.
+    try {
+      const json = JSON.parse(text) as { message?: string; error?: string };
+      throw new Error(json.message ?? json.error ?? `HTTP ${res.status}`);
+    } catch (inner) {
+      if (inner instanceof SyntaxError) throw new Error(`HTTP ${res.status}`);
```

ESLint

[error] There is no `cause` attached to the symptom error being thrown.

(preserve-caught-error)